.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,93 @@
+name: Pre-commit actions for Delphix Virtualization Platform SDK
+
+on: [pull_request]
+
+jobs:
+  pytest:
+    name: Test ${{ matrix.package }} on ${{ matrix.os }} using pytest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [2.7]
+        os: [ubuntu-latest, macos-latest]
+        package: [common, libs, platform, tools]
+
+    steps:
+    - name: Checkout ${{ matrix.package }} project
+      uses: actions/checkout@v1
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install ${{ matrix.package }} dependencies
+      working-directory: ${{ matrix.package }}
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt --index-url https://pypi.org/simple --extra-index-url https://test.pypi.org/simple
+
+    - name: Install ${{ matrix.package }} project in editable mode
+      working-directory: ${{ matrix.package }}
+      run: |
+        pip install -e . --index-url https://pypi.org/simple --extra-index-url https://test.pypi.org/simple
```

Comment:
nit: we don't need to install it in editable mode. I think we can just run `pip install .` at this point.